src/Faker/Provider/pl_PL/LicensePlate.php

```diff
@@ -0,0 +1,571 @@
+            'pyrzycki' => ['PY'],
+            'sławieński' => ['SL'],
+            'stargardzki' => ['ST'],
+            'szczecinecki' => ['SZ'],
+            'świdwiński' => ['SD'],
+            'wałecki' => ['WA'],
+        ],
+        'U' => [
+            'Siły Zbrojne Rzeczypospolitej Polskiej' => ['A', 'B', 'C', 'D', 'E', 'G', 'I', 'J', 'K', 'L'],
+        ],
+        'H' => [
+            'Centralne Biuro Antykorupcyjne' => ['A'],
+            'Służba Ochrony Państwa' => ['BA', 'BB', 'BE', 'BF', 'BG'],
+            'Służba Celno-Skarbowa' => ['CA', 'CB', 'CC', 'CD', 'CE', 'CF', 'CG', 'CH', 'CJ', 'CK', 'CL', 'CM', 'CN', 'CO', 'CP', 'CR'],
+            'Agencja Bezpieczeństwa Wewnętrznego' => ['K'],
+            'Agencja Wywiadu' => ['K'],
+            'Służba Kontrwywiadu Wojskowego' => ['M'],
+            'Służba Wywiadu Wojskowego' => ['M'],
+            'Policja' => ['PA', 'PB', 'PC', 'PD', 'PE', 'PF', 'PG', 'PH', 'PJ', 'PK', 'PL', 'PL', 'PL', 'PL', 'PL', 'PM', 'PN', 'PP', 'PS', 'PT', 'PU', 'PW', 'PZ'],
+            'Straż Graniczna' => ['WA', 'WK'],
+        ],
+    ];
+
+    /**
+     * Generates random license plate.
+     * Generation method based on the description from https://pl.wikipedia.org/wiki/Tablice_rejestracyjne_w_Polsce#Tablice_standardowe
+     *
+     * @param  array|null $voivodeships
+     * @param  array|null $counties
+     *
+     * @return string
+     */
+    public function licensePlate($voivodeships = null, $counties = null)
+    {
+        if (is_array($voivodeships)) {
+            $availableVoivodeships = array_keys(static::$voivodeships);
+            $voivodeships = array_filter($voivodeships, function ($value) use ($availableVoivodeships) {
+                return in_array($value, $availableVoivodeships);
+            });
+        } else {
+            $voivodeships = [];
+        }
+
+        if (count($voivodeships)) {
+            $voivodeship = static::randomElement($voivodeships);
+
+            $voivodeshipCode = static::$voivodeships[$voivodeship];
+        } else {
+            $voivodeshipCode = static::randomElement(static::$voivodeships);
+        }
+
+        if (is_array($counties)) {
+            $availableCounties = array_keys(static::$counties[$voivodeshipCode]);
+            $counties = array_filter($counties, function ($value) use ($availableCounties) {
+                return in_array($value, $availableCounties);
+            });
+        } else {
+            $counties = [];
+        }
+
+        if (count($counties)) {
+            $countyCodes = static::$counties[$voivodeshipCode][static::randomElement($counties)];
+        } else {
+            $countyCodes = static::randomElement(static::$counties[$voivodeshipCode]);
+        }
+
+        $countyCode = static::randomElement($countyCodes);
+
+        if (strlen($countyCode) === 1) {
+            switch (static::numberBetween(1, 5)) {
+                case 1:
+                    $randomCode = static::regexify('\d{5}');
```

Comment:
You can store the formats in a property of this class and then choose it using `randomElement` (see `companyName`)